crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -788,6 +788,7 @@ def proposal_status
           ).first
         ]
       end
+      result.compact!
```

Comment:
Is it possible that `result` is empty? If yes, compact will return `nil` in this case which breaks the loop in the next line.